dsp_permissions_scripts/models/group.py

```diff
@@ -2,53 +2,108 @@
 
 import re
 from typing import Any
+from typing import Self
+from typing import TypeAlias
+from typing import Union
 
 from pydantic import BaseModel
 from pydantic import ConfigDict
+from pydantic import Discriminator
+from pydantic import Tag
 from pydantic import model_validator
+from typing_extensions import Annotated
 
 from dsp_permissions_scripts.models.errors import InvalidGroupError
-from dsp_permissions_scripts.models.errors import InvalidIRIError
 from dsp_permissions_scripts.utils.helpers import KNORA_ADMIN_ONTO_NAMESPACE
 
 PREFIXED_IRI_REGEX = r"^[\w-]+:[\w -]+$"
 NAMES_OF_BUILTIN_GROUPS = ["SystemAdmin", "Creator", "ProjectAdmin", "ProjectMember", "KnownUser", "UnknownUser"]
 
 
-class Group(BaseModel):
+class BuiltinGroup(BaseModel):
+    """Represents a DSP-builtin group, in the form 'knora-admin:ProjectAdmin'"""
+
     model_config = ConfigDict(frozen=True)
 
     prefixed_iri: str
 
-    @model_validator(mode="before")
-    @classmethod
-    def _shorten_iri(cls, data: Any) -> Any:
-        if not isinstance(data, dict):
-            return data
-        data["prefixed_iri"] = data["prefixed_iri"].replace(KNORA_ADMIN_ONTO_NAMESPACE, "knora-admin:")
-        return data
+    @model_validator(mode="after")
+    def _check_regex(self) -> Self:
```

Comment:
detail: technically, this is not a regex, check, right?

Reply:
Right.